src/background/stores/card-selection-store.ts

```diff
@@ -24,17 +24,72 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
 
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
-            rules: null,
+            rules: {},
         };
 
         return defaultValue;
     }
 
-    public toggleRuleExpandCollapse(payload: RuleExpandCollapsePayload): void {}
+    private deselectAllCardsInRule = (rule: RuleExpandCollapseData): void => {
+        if (!rule) {
+            return;
+        }
 
-    public toggleCardSelection(payload: CardSelectionPayload): void {}
+        for (const resultInstanceUid of Object.keys(rule.cards)) {
+            rule.cards[resultInstanceUid] = false;
+        }
+    };
 
-    public CollapseAllRules(): void {}
+    public toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
+        if (!payload || !payload.ruleId || !this.state.rules || !this.state.rules[payload.ruleId]) {
+            return;
+        }
 
-    private onScanCompleted = (payload: UnifiedScanCompletedPayload): void => {};
+        const rule = this.state.rules[payload.ruleId];
+
+        rule.isExpanded = !rule.isExpanded;
+
+        if (!rule.isExpanded) {
+            this.deselectAllCardsInRule(rule);
+        }
+
+        super.emitChanged();
+    };
+
+    public toggleCardSelection = (): void => {};
+
+    public CollapseAllRules = (): void => {
+        for (const ruleId of Object.keys(this.state.rules)) {
```

Comment:
I think this would be a good place to use ```forOwn``` from lodash.

Reply:
I was looking for something like forOwn, but I didn't know what it was. Thanks!